react-file-uploads/client/src/UploadForm.tsx

```diff
@@ -0,0 +1,49 @@
+import { type FormEvent, useState } from 'react';
+
+type Image = {
+  imageId: number;
+  url: string;
+  caption: string;
+};
+
+export function UploadForm() {
+  const [imageFile, setImageFile] = useState<Image>();
+
+  async function handleSubmit(event: FormEvent<HTMLFormElement>) {
+    try {
+      event.preventDefault();
+      const eventTarget = event.target as HTMLFormElement;
```

Comment:
`event.currentTarget` has the proper type (so there's no need for the type assertion) and is essentially identical to `event.target`.